src/raster_utils/helpers.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from shapely.geometry import Polygon
+
+if TYPE_CHECKING:
+    import xarray
+
+
+def get_raster_bounds(xarr: xarray.DataArray) -> Polygon:
```

Comment:
This whole function can be simplified with:

```python
import rioxarray
from shapely.geometry import box


# Get the bounds (bbox) using rioxarray with rio accessor
bbox = data.rio.bounds()  # (minx, miny, maxx, maxy)

# Create a shapely Polygon from the bounding box using the box() function
minx, miny, maxx, maxy = bbox
polygon = box(minx, miny, maxx, maxy)
```

We already have `rioxarray` as dependency.

Reply:
Up to you if you want to simplify it :stuck_out_tongue:

Reply:
That is a good remark, thank you for spotting it. Updated.